flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/BackPressureITCase.java

```diff
@@ -0,0 +1,165 @@
+import org.apache.flink.runtime.rest.handler.legacy.backpressure.OperatorBackPressureStats;
+import org.apache.flink.runtime.rest.handler.legacy.backpressure.OperatorBackPressureStatsResponse;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.test.util.BlockingSink;
+import org.apache.flink.test.util.IdentityMapFunction;
+import org.apache.flink.test.util.InfiniteIntegerSource;
+import org.apache.flink.util.TestLogger;
+import org.apache.flink.util.function.SupplierWithException;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+
+import static org.apache.flink.runtime.testutils.CommonTestUtils.waitUntilCondition;
+
+/**
+ * Integration test for operator back pressure tracking.
+ */
+public class BackPressureITCase extends TestLogger {
+
+	private static final JobID TEST_JOB_ID = new JobID();
+	private static final int NUM_TASKS = 3;
+	private static final int BACK_PRESSURE_REQUEST_INTERVAL_MS = 5;
+	private static final int TASKS_BECOMING_BACK_PRESSURED_TIMEOUT_MS = 15 * 1000;
+	private static final int MAX_BACK_PRESSURE_RATIO = 1;
+
+	private TestingMiniCluster testingMiniCluster;
+	private DispatcherGateway dispatcherGateway;
+
+	@Before
+	public void setUp() throws Exception {
+		final Configuration configuration = new Configuration();
+		configuration.addAll(createBackPressureSamplingConfiguration());
+		configuration.addAll(createNetworkBufferConfiguration());
+
+		final TestingMiniClusterConfiguration testingMiniClusterConfiguration = new TestingMiniClusterConfiguration.Builder()
+			.setNumSlotsPerTaskManager(NUM_TASKS)
+			.setConfiguration(configuration)
+			.build();
+
+		testingMiniCluster = new TestingMiniCluster(testingMiniClusterConfiguration);
+		testingMiniCluster.start();
+		dispatcherGateway = testingMiniCluster.getDispatcherGatewayFuture().get();
+	}
+
+	private static Configuration createBackPressureSamplingConfiguration() {
+		final Configuration configuration = new Configuration();
+		configuration.setInteger(WebOptions.BACKPRESSURE_REFRESH_INTERVAL, 1000);
+		configuration.setInteger(WebOptions.BACKPRESSURE_NUM_SAMPLES, 1);
+		configuration.setInteger(WebOptions.BACKPRESSURE_CLEANUP_INTERVAL, Integer.MAX_VALUE);
+		return configuration;
+	}
+
+	private static Configuration createNetworkBufferConfiguration() {
+		final Configuration configuration = new Configuration();
+
+		final int memorySegmentSizeKb = 32;
+		final String networkBuffersMemory = (memorySegmentSizeKb * NUM_TASKS) + "kb";
+
+		configuration.setString(TaskManagerOptions.MEMORY_SEGMENT_SIZE, memorySegmentSizeKb + "kb");
+		configuration.setString(NettyShuffleEnvironmentOptions.NETWORK_BUFFERS_MEMORY_MIN, networkBuffersMemory);
+		configuration.setString(NettyShuffleEnvironmentOptions.NETWORK_BUFFERS_MEMORY_MAX, networkBuffersMemory);
+		return configuration;
+	}
+
+	@Test
```

Comment:
Just a control question: was this test failing before your fix?

Reply:
yes it fails without my fix